app/routers/wallet.py

```diff
@@ -0,0 +1,314 @@
+        logger.info(f"accept_taa_response: {accept_taa_response}")
+        if accept_taa_response != {}:
+            error_json = accept_taa_response.json()
+            raise HTTPException(
+                status_code=404,
+                detail=f"Something went wrong. Could not accept TAA. {error_json}",
+            )
+        assign_pub_did_response = await aries_agent_controller.wallet.assign_public_did(
+            did_object["did"]
+        )
+        logger.info(f"assign_pub_did_response:\n{assign_pub_did_response}")
+        if (
+            not assign_pub_did_response["result"]
+            or assign_pub_did_response["result"] == {}
+        ):
+            error_json = assign_pub_did_response.json()
+            raise HTTPException(
+                status_code=500,
+                detail=f"Something went wrong.\nCould not assign DID. {error_json}",
+            )
+        get_pub_did_response = await aries_agent_controller.wallet.get_public_did()
+        logger.info(f"get_pub_did_response:\n{get_pub_did_response}")
+        if not get_pub_did_response["result"] or get_pub_did_response["result"] == {}:
+            error_json = get_pub_did_response.json()
+            raise HTTPException(
+                status_code=404,
+                detail=f"Something went wrong. Could not obtain public DID. {error_json}",
+            )
+        issuer_nym = get_pub_did_response["result"]["did"]
+        issuer_verkey = get_pub_did_response["result"]["verkey"]
+        issuer_endpoint = await aries_agent_controller.ledger.get_did_endpoint(
+            issuer_nym
+        )
+        if not issuer_endpoint:
+            raise HTTPException(
+                status_code=404,
+                detail="Something went wrong. Could not obtain issuer endpoint.",
+            )
+        issuer_endpoint_url = issuer_endpoint["endpoint"]
+        final_response = DidCreationResponse(
+            did_object=did_object,
+            issuer_verkey=issuer_verkey,
+            issuer_endpoint=issuer_endpoint_url,
+        )
+        await aries_agent_controller.terminate()
+        return final_response
+    except Exception as e:
+        await aries_agent_controller.terminate()
+        logger.error(f"The following error occured:\n{e!r}")
+        raise HTTPException(
+            status_code=500,
+            detail=f"Something went wrong: {e!r}",
+        )
+
+
+@router.get("/", tags=["wallets"])
+async def wallets_root():
+    """
+    The default endpoints for wallets
+
+    TODO: Determine what this should return or
+    whether this should return anything at all
+    """
+    return {
+        "message": "Wallets endpoint. Please, visit /docs to consult the Swagger docs."
+    }
+
+
+# TODO: This should be somehow retsricted?!
+@router.post("/create-wallet", tags=["wallets"])
+async def create_wallet(wallet_payload: dict = None):
+    """
+    Create a new wallet
+
+    Parameters:
+    -----------
+
```

Comment:
I think this should mention which attributes are required. @morrieinmaas
Maybe mention that the wallet name should be unique as well.

Reply:
hmm, well for now it's a dict really - there is an example in the docstring. I reckon that's a good place to use a schema (model) instead